File: src/gui/window_flags.rs
use bitflags::bitflags;

bitflags! 
{
    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
    pub struct WindowFlags: u32
    {
        const None = 0;
        const Default = Self::Shown.bits() | Self::Resizable.bits();
        const Fullscreen = 2 << 1;
        // const FullscreenDesktop;
        // const Opengl;
        // const Vulkan;
        const Shown = 2 << 5;
        const Hidden = 2 << 6;
        const Borderless = 2 << 7;
        const Resizable = 2 << 8;
        const Minimized = 2 << 9;
        const Maximized = 2 << 10;
        // const InputGrabbed;
        // const InputFocus;
        // const MouseFocus;
        // const Foreign;
        // const AllowHighdpi;
        // const MouseCapture;
        // const AlwaysOnTop;
        // const SkipTaskbar;
        // const Utility;
        // const Tooltip;
        // const PopupMenu;
    }
}

File: src/gui/window_flags.zig
pub const WindowFlags = packed struct(u32) {
    None: bool = false,
    Fullscreen: bool = false,
    // FullscreenDesktop,
    // Opengl,
    // Vulkan,
    Shown: bool = false,
    Hidden: bool = false,
    Borderless: bool = false,
    Resizable: bool = false,
    Minimized: bool = false,
    Maximized: bool = false,
    // InputGrabbed,
    // InputFocus,
    // MouseFocus,
    // Foreign,
    // AllowHighdpi,
    // MouseCapture,
    // AlwaysOnTop,
    // SkipTaskbar,
    // Utility,
    // Tooltip,
    // PopupMenu,
    _padding: u24 = 0,

    const Self = @This();
    pub fn default() Self {
        return WindowFlags { .Shown = true, .Resizable = true } ;
    }

    pub fn as_u32(self: Self) u32
    {
        return @bitCast(u32, self);
    }
};

const std = @import("std");
test "window flags" {
    std.debug.print("Enum: 0x{x}\n", .{WindowFlags.default().as_u32()});
}
